### ISSUE TITLE
mysql-cdc  MySQL user doesn't have enough privileges

### ISSUE THREAD

**Issue by SparkLover** [risingwavelabs/risingwave#15757]
### Describe the bug

source cannot pass validation: INVALID_ARGUMENT: MySQL user doesn't have enough privileges: [RELOAD, SHOW DATABASES, REPLICATION SLAVE, REPLICATION CLIENT, SELECT]

### Error message/log

```text
24-03-18 21:12:34] [XX000] ERROR: Failed to execute the statement
[2024-03-18 21:12:34] Caused by these errors (recent errors listed first):
[2024-03-18 21:12:34] 1: gRPC request to meta service failed: Internal error
[2024-03-18 21:12:34] 2: failed to create source worker
[2024-03-18 21:12:34] 3: failed to create SplitEnumerator
[2024-03-18 21:12:34] 4: source cannot pass validation: INVALID_ARGUMENT: MySQL user doesn't have enough privileges: [RELOAD, SHOW DATABASES, REPLICATION SLAVE, REPLICATION CLIENT, SELECT]
```

### To Reproduce

CREATE TABLE orders (
order_id int,
price decimal,
PRIMARY KEY (order_id)
) WITH (
connector = 'mysql-cdc',
hostname = 'localhost',
port = '3306',
username = 'root',
password = '123456',
database.name = 'db01',
table.name = 'orders'
);

### Expected behavior

I give root user all permissions
<img width="567" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/44989650/34182ff6-e5b8-4166-99ef-1702bafe6f66">

### How did you deploy RisingWave?

docker 

### The version of RisingWave

PostgreSQL 9.5.0-RisingWave-1.7.1 (3a545de8b9a46b80fb3172ba406cd5e5a74ce540)

**Comment by lmatz** [comment]
Hi @SparkLover 

are there enough permissions granted to the corresponding user in MySQL? https://docs.risingwave.com/docs/current/ingest-from-mysql-cdc/#set-up-mysql:
> CREATE USER 'user'@'%' IDENTIFIED BY 'password';
> GRANT SELECT, RELOAD, SHOW DATABASES, REPLICATION SLAVE, REPLICATION CLIENT ON *.* TO 'user'@'%';
> FLUSH PRIVILEGES;

**Comment by SparkLover** [comment]
thanks ：
I can use it after creating a new user according to the official website's method. It's strange that it can't work if I use the root user.
















At 2024-03-18 21:30:27, "lmatz" ***@***.***> wrote:

Hi @SparkLover

are there enough permissions granted to the corresponding user in MySQL? https://docs.risingwave.com/docs/current/ingest-from-mysql-cdc/#set-up-mysql:

CREATE USER 'user'@'%' IDENTIFIED BY 'password';
GRANT SELECT, RELOAD, SHOW DATABASES, REPLICATION SLAVE, REPLICATION CLIENT ON . TO 'user'@'%';
FLUSH PRIVILEGES;

**Comment by lmatz** [comment]
cc: @StrikeW @cyliu0   if 
> I can use it after creating a new user according to the official website's method. It's strange that it can't work if I use the root user.

is expected?

**Comment by lmatz** [comment]
Hi @SparkLover , what version of MySQL are you using?

**Comment by StrikeW** [comment]
Hi @SparkLover, I think in v1.7.2 we have fixed the issue with `GRANT ALL`.  You may try out the v1.7.2 version.

**Comment by SparkLover** [comment]
thanks:
mysql 5.7
















At 2024-03-19 12:56:28, "lmatz" ***@***.***> wrote:

Hi @SparkLover , what version of MySQL are you using?

**Comment by SparkLover** [comment]
thanks ，I'll try v1.7.2 versions
















At 2024-03-19 13:13:59, "StrikeW" ***@***.***> wrote:

Hi @SparkLover, I think in v1.7.2 we have fixed the issue with GRANT ALL. You may try out the v1.7.2 version.

**Comment by StrikeW** [comment]
Sorry, the fix PR
#15395 didn't cherry-picked into v1.7.2 yet
We didn't consider the GRANT ALL for mysql 5.7 before, the fix has merged in main and will go into v1.8.
